_ramblings/2020-04-02-keep-it-to-yourself.md

```diff
@@ -94,6 +195,36 @@ control access to individual hosts.
 - TODO: Implement this.
 - TODO: Note it's stored in `pass`, more on below.
 
+Resident "discoverable" keys can be stored on a security key which supports the
+FIDO standard.
+
+**Generate an SSH key as a resident FIDO key:**
+The `-t ${KEY_TYPE}-sk` tells `ssh-keygen` to use the smart card to generate
+the key.
+```sh
+NAME=mykey
+KEY_TYPE=ed25519
+ssh-keygen \
+   -f ~/.ssh/id_${KEY_TYPE}_sk_rk_${NAME} \
+   -t ${KEY_TYPE}-sk \
+   -O resident \
+   -O application=ssh:${NAME} \
+   -O verify-required
+```
+
+**List FIDO keys:**
+```sh
+ykman fido credentials list
+```
+The "RP ID" should be `ssh:...` as you set above. You can ignore the user and
+display names.
+
+**Recover resident SSH keys from FIDO SK:**
+```sh
+cd ~/.ssh
+ssh-keygen -K
+```
```

Comment:
Recovered key seems to have a default note of `username@hostname` which in this case I don't want.